.github/workflows/slsa-goreleaser.yml

```diff
@@ -0,0 +1,163 @@
+          echo "commit=$GITHUB_SHA" >> "$GITHUB_OUTPUT"
+          echo "version=$(git describe --tags --always --dirty | cut -c2-)" >> "$GITHUB_OUTPUT"
+          echo "tree-state=$(if git diff --quiet; then echo "clean"; else echo "dirty"; fi)" >> "$GITHUB_OUTPUT"
+
+  # Generate the SBOM file and upload it to release
+  generate-sbom:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write # To upload assets to release.
+    needs: args
+    outputs:
+      hash: ${{ steps.sbom-hash.outputs.hash }}
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3
+      - name: Install SYFT
+        id: install-syft
+        uses: anchore/sbom-action/download-syft@06e109483e6aa305a2b2395eabae554e51530e1d # v0.13.1
+      - name: Generate SBOM
+        run: syft . -o json --file helm-sigstore-${{needs.args.outputs.version}}-sbom.json
+      - name: Get the Hash to SBOM
+        id: sbom-hash
+        run: echo "hash=$(sha256sum helm-sigstore-${{needs.args.outputs.version}}-sbom.json | base64 -w0)" >> "$GITHUB_OUTPUT"
+      - name: Upload SBOM artifact
+        uses: actions/upload-artifact@83fd05a356d7e2593de66fc9913b3002723633cb # v3.1.1
+        with:
+          name: helm-sigstore-${{needs.args.outputs.version}}-sbom.json
+          path: helm-sigstore-${{needs.args.outputs.version}}-sbom.json
+          if-no-files-found: error
+          retention-days: 5
+      - name: Release SBOM
+        uses: softprops/action-gh-release@de2c0eb89ae2a093876385947365aca7b0e5f844 # v0.1.15
+        id: release-tag-with-sbom
+        with:
+          files: helm-sigstore-${{needs.args.outputs.version}}-sbom.json
+
+  # Generate and Upload SBOM Provenance to release
+  sbom-provenance:
+    needs: generate-sbom
+    permissions:
+      actions: read # Needed for detection of GitHub Actions environment.
+      id-token: write # Needed for provenance signing and ID
+      contents: write # Needed for release uploads
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.4.0
+    with:
+      base64-subjects: "${{ needs.generate-sbom.outputs.hash }}"
+      upload-assets: true
+
+  # Build
+  release:
+    permissions:
+      id-token: write # To sign the provenance.
+      contents: write # To upload assets to release.
+      actions: read # To read the workflow path.
+    strategy:
+      fail-fast: false
+      matrix:
+        goos: [linux, darwin, windows]
+        goarch: [amd64, arm64]
+        include:
+          - goos: linux
+            goarch: s390x
+          - goos: linux
+            goarch: ppc64le
+          - goos: linux
+            goarch: arm
+        exclude:
+          - goos: windows
+            goarch: arm64
+    needs: args
+    uses: slsa-framework/slsa-github-generator/.github/workflows/builder_go_slsa3.yml@v1.4.0
```

Comment:
looks very cool, but i feel it added a lot of maintenance, is it possible to do similar implementation like we did here: https://github.com/sigstore/policy-controller/blob/main/.github/workflows/release.yaml ?

Reply:
You mean using the generic provenance generator? Sure, it it possible, I've done this way to suggest the workflow using the SLSA 3 Go Builder, but it is also possible to just generate the provenance in a different step using the generator_generic_slsa3.

I'll change to it then.

Reply:
@cpanato I'm curious what the added maintenance burden you encountered was. We are considering creating goreleaser based workflows in the future and would appreciate the input.